libfabric.spec.in

```diff
@@ -8,10 +8,15 @@ Url: http://www.github.com/ofiwg/libfabric
 Source: http://www.openfabrics.org/downloads/fabrics/%{name}-%{version}.tar.bz2
 Prefix: ${_prefix}
 
-BuildRequires: libnl-devel
+%if 0%{?configopts:1}
+# skip BuildRequires for custom build
```

Comment:
Not without maintaining a map of provider : package names.

Bear in mind though we don't completely toss the dependency checking away: the ./configure script will still  print out an error if anything is missing (vs rpmbuild printing the same thing and bailing right away).